backend/ethereum/channel/funder.go

```diff
@@ -211,7 +213,7 @@ func (f *Funder) fundAssets(ctx context.Context, channelID channel.ID, req chann
 // funding request. It is idempotent.
 func (f *Funder) sendFundingTx(ctx context.Context, request channel.FundingReq, contract assetHolder, fundingID [32]byte) (txs []*types.Transaction, fatal error) {
 	bal := request.Agreement[contract.assetIndex][request.Idx]
-	// nolint: gocritic
+	//nolint:gocritic
```

Comment:
This `//nolint` can be avoided by improving the structure:

```go
// sendFundingTx sends and returns the TXs that are needed to fulfill the
// funding request. It is idempotent.
func (f *Funder) sendFundingTx(ctx context.Context, request channel.FundingReq, contract assetHolder, fundingID [32]byte) (txs []*types.Transaction, fatal error) {
	bal := request.Agreement[contract.assetIndex][request.Idx]

	if bal == nil || bal.Sign() <= 0 {
		f.log.WithFields(log.Fields{"channel": request.Params.ID(), "idx": request.Idx}).Debug("Skipped zero funding.")
		return nil, nil
	}

	alreadyFunded, err := f.checkFunded(ctx, bal, contract, fundingID)
	if err != nil {
		return nil, errors.WithMessage(err, "checking funded")
	}

	if alreadyFunded {
		f.log.WithFields(log.Fields{"channel": request.Params.ID(), "idx": request.Idx}).Debug("Skipped second funding.")
		return nil, nil
	}

	return f.deposit(ctx, bal, wallet.Address(*contract.Address), fundingID)
}
```

Reply:
put it in https://github.com/hyperledger-labs/go-perun/pull/260/commits/cdc57f576b60eb39e266ad6c3200d500c519d2f3